yarn-project/acir-simulator/src/acvm/acvm.ts

```diff
@@ -29,6 +29,7 @@ export const ONE_ACVM_FIELD: ACVMField = `0x${'00'.repeat(Fr.SIZE_IN_BYTES - 1)}
  * The supported oracle names.
  */
 type ORACLE_NAMES =
+  | 'computeSelector'
```

Comment:
despite the convinience, computing the selector is something that should be done at compile time, and not at runtime. A malicious app could alter the source code / calls being made without changing the verification key.

Reply:
In the sandbox a malicious actor can do whatever they want. Calling other functions are also done through oracles where it could just return success independent of what happens. This is essentially just a helper to compute an input to another function that is completely controlled by a possibly malicious app so not sure it changes much there 🤷

Reply:
I may be missing something, but I agree with @Maddiaa0 on this one. Calling other functions as you say can also return anything, but we then have a kernel circuit that ensures that the oracle behaved properly. But it's true it doesn't affect the security of sandbox as it is today.

I'm fine merging this, but we should add an issue to replace it with a Noir-only implementation in the future (ie before testnet).